src/content/doc-surrealql/datamodel/idioms.mdx

```diff
@@ -0,0 +1,601 @@
+
+## Parts of an Idiom
+
+An idiom is made up of one or more **parts**, each of which can be one of several types:
+
+- [**Field**](#field-access): Access a field by name.
+- [**Index**](#index-access): Access an element of an array by its index.
+- [**All**](#all-elements): Access all elements or fields.
+- [**Last**](#last-element): Access the last element of an array.
+- **First**: Access the first element of an array.
+- [**Flatten**](#flatten): Flatten nested arrays.
+- [**Where**](#where-filter): Filter elements based on a condition.
+- [**Method**](#method-chaining): Call a method on the current data.
+- [**Graph**](#graph-navigation): Navigate through graph relationships.
+- [**Destructure**](#destructuring): Destructure nested objects.
+- [**Optional**](#optional-parts): Indicate that the following part is optional.
+
+In this section, we'll explore each part in detail with examples to help you understand how to use idioms in SurrealQL.
+
+### Field Access
+
+Since SurrealDB is, at its core, a document database each record is stored on an underlying key-value store storage engine, with the ability to store arbitrary arrays, objects, and many other types of data.  you can access fields in a record using the dot notation. To access a field in an object, use a dot `.` followed by the field name. 
+
+This is mostly helpful when accessing fields within a record, but can also be used to access fields within an array.
+
+For example, using the `CREATE` statement to  a record into the `person` table:
+
+
+```surql title="Query"
+CREATE person CONTENT {
+    name: "John Doe",
+    age: 30,
+    address: {
+      city: "New York",
+      country: "USA"
+    }
+};
+```
+```json title="Response"
+{
+  "person": {
+    "name": "John Doe",
+    "age": 30,
+    "address": {
+      "city": "New York",
+      "country": "USA"
+    }
+  }
+}
+```
+
+To access the `city` field within the `address` object, you can use the following idiom:
+
+```surql title="Query"
+SELECT address.city FROM person;
+```
+
+```json title="Response"
+[
+  {
+    "address": {
+      "city": "New York"
+    }
+  }
+]
+```
+In this example, `person.name` is an idiom that accesses the `name` field of the `person` record.
+
+### Index Access
+
+To access an element in an array by its index, use square brackets `[]` with the index inside. For example let's say we have a `school` record with some student results. 
+
+```surql title="Query"
+CREATE student SET results = [
+	{ score: 76, date: "2017-06-18T08:00:00Z", name: "Algorithmics" },
+	{ score: 83, date: "2018-03-21T08:00:00Z", name: "Concurrent Programming" },
+	{ score: 69, date: "2018-09-17T08:00:00Z", name: "Advanced Computer Science 101" },
+	{ score: 73, date: "2019-04-20T08:00:00Z", name: "Distributed Databases" },
+];
+```
+
+```json title="Response"
```

Comment:
Okay, I guess I have my answer about what happens when the `json` attribute is used but the output is not actual JSON 😆 The linter gets upset and highlights everything in red and the actual output is pretty wild

![image](https://github.com/user-attachments/assets/ff3c4478-2659-4778-a7aa-4987c879740a)

Reply:
I'll keep them all as bash